pkg/clustermesh/mcsapi/serviceimport_controller.go

```diff
@@ -0,0 +1,535 @@
+			}
+
+			foundPortByName, hasConflict, conflictDuplicatedPortName := checkDuplicatedPortNameConflict(portMergeValue, portsByName)
+			if conflict == "" {
+				conflict = conflictDuplicatedPortName
+			}
+			if hasConflict {
+				continue
+			}
+
+			portIndex := slices.IndexFunc(ports, func(currPort portMerge) bool {
+				return isSimilarPort(port, currPort.ServicePort)
+			})
+			if portIndex == -1 {
+				if !foundPortByName {
+					// We don't override any port in portsByName so that this map
+					// always keep the values from the oldest service export
+					// exporting that port name
+					portsByName[port.Name] = portMergeValue
+				}
+				ports = append(ports, portMergeValue)
+			} else if conflict == "" {
+				conflict = checkPortConflict(portMergeValue, ports[portIndex])
+			}
+		}
+	}
+	return ports, conflict
+}
+
+func mergedPortsToMCSPorts(mergedPorts []portMerge) []mcsapiv1alpha1.ServicePort {
+	ports := make([]mcsapiv1alpha1.ServicePort, 0, len(mergedPorts))
+	for _, port := range mergedPorts {
+		ports = append(ports, port.ServicePort)
+	}
+	return ports
+}
+
+func getServiceImportStatus(svcExportByCluster operator.ServiceExportsByCluster) mcsapiv1alpha1.ServiceImportStatus {
+	clusters := []mcsapiv1alpha1.ClusterStatus{}
+	clusterNames := slices.Collect(maps.Keys(svcExportByCluster))
+	slices.Sort(clusterNames)
+	for _, cluster := range clusterNames {
+		clusters = append(clusters, mcsapiv1alpha1.ClusterStatus{
+			Cluster: cluster,
+		})
+	}
+	return mcsapiv1alpha1.ServiceImportStatus{Clusters: clusters}
+}
+
+// checkConflictExport check if there are any conflict to be added on
+// the ServiceExport object. This function does not check for conflict on the
+// ports field this aspect should be done by mergePorts
+func checkConflictExport(svcExportByCluster operator.ServiceExportsByCluster, oldestCluster string) string {
+	orderedSvcExports := orderSvcExportByPriority(svcExportByCluster)
+	clusterCount := len(orderedSvcExports)
+
+	getTypeFunc := func(svcSpec *mcsapitypes.MCSAPIServiceSpec) string {
+		return string(svcSpec.Type)
+	}
+	getSessionAffinityFunc := func(svcSpec *mcsapitypes.MCSAPIServiceSpec) string {
+		return string(svcSpec.SessionAffinity)
+	}
+	getSessionAffinityConfigFunc := func(svcSpec *mcsapitypes.MCSAPIServiceSpec) string {
+		if svcSpec.SessionAffinityConfig == nil ||
+			svcSpec.SessionAffinityConfig.ClientIP == nil ||
+			svcSpec.SessionAffinityConfig.ClientIP.TimeoutSeconds == nil {
+			return ""
+		}
+		return string(*svcSpec.SessionAffinityConfig.ClientIP.TimeoutSeconds)
+	}
+	getterFuncs := []struct {
+		name       string
+		getterFunc func(svcSpec *mcsapitypes.MCSAPIServiceSpec) string
+	}{
+		{
+			name: "type", getterFunc: getTypeFunc,
+		},
+		{
+			name: "sessionAffinity", getterFunc: getSessionAffinityFunc,
+		},
+		{
+			name: "sessionAffinityConfig.clientIP", getterFunc: getSessionAffinityConfigFunc,
+		},
+	}
+	for i, svcExport := range orderedSvcExports {
```

Comment:
```suggestion
	for i, svcExport := range orderedSvcExports[1:] {
```

To avoid checking the first, which is always the oldest.